app/controllers/users_controller.rb

```diff
@@ -11,32 +11,21 @@ def index
 
   def create
     @current_team_users = current_team_users
-    @user = User.find_or_create_by(email: allowed_params[:email]) do |u|
-      # Set to true if the user was not found
-      u.was_created = true
-      u.name        = allowed_params[:name]
-      u.initials    = allowed_params[:initials]
-      u.username    = allowed_params[:username]
-    end
-    authorize @user
-
-    if @user.new_record? && !@user.save
-      render 'index'
-      return
-    end
+    @user = User.find_by(email: allowed_params[:email])
 
-    if policy_scope(User).include?(@user)
-      flash[:alert] = I18n.t('is already a member of this project', scope: 'users', email: @user.email)
-    else
-      policy_scope(User) << @user
-      @user.teams << current_team unless @user.teams.include?(current_team)
-      if @user.was_created
-        flash[:notice] = I18n.t('was sent an invite to join this project', scope: 'users', email: @user.email)
+    if @user
```

Comment:
Well, it seems to be doing a lot of things into the controller. What about a service class which receives an email and tries to do these things? As result it could return true or false, and the message could be get from its instance, something like:

```ruby
def create
  @current_team_users = current_team_users
  authorize @current_team_users
  
  @user = User.find_by(email: allowed_params[:email])
  authorize @user

  project_enroller = ProjectEnroller.new(@user)
  if project_enroller.enroll
    flash[:notice] = project_enroller.message
  else
    flash[:alert] = project_enroller.message
  end

  ...
```